### PR TITLE
Allow image as input

### LINKED ISSUE #113: [ENHANCEMENT] Allow image as input
Allow image as input of `MapieClassifier`: if we create model scikit-learn compatible model for image classification for instance, we should be able to have `X` as a matrix of images

### PR DESCRIPTION
**Pull request by vincentblot28**
# Description

Add possibility to have image as input

Fixes #113 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: mapie/classification.py [hunk 1]
```diff
@@ -222,7 +223,13 @@ def _check_estimator(
             If the estimator is not fitted and ``cv`` attribute is "prefit".
         """
         if estimator is None:
-            return LogisticRegression(multi_class="multinomial").fit(X, y)
+            if not self.image_input:
+                return LogisticRegression(multi_class="multinomial").fit(X, y)
+            else:
+                raise ValueError(
+                    "LogisticRegression's input can't be an image."
```
Comment (gmartinonQM) [inline mapie/classification.py:230]: Ajouter "Default Logistic Regression..."

File: mapie/utils.py [hunk 2]
```diff
@@ -348,3 +348,27 @@ def check_nan_in_aposteriori_prediction(X: ArrayLike) -> None:
             + "belongs to every resamplings.\n"
             "Increase the number of resamplings"
         )
+
+
+def input_is_image(X: ArrayLike) -> None:
```
Comment (gmartinonQM) [inline mapie/utils.py:353]: Peut-on renommer en "check_input_is_image" ?

File: mapie/tests/test_classification.py [hunk 3]
```diff
@@ -275,6 +328,51 @@ def test_no_fit_predict() -> None:
...
+        method="cumulated_score",
+        cv="prefit",
+        random_state=42
+    )
+    mapie.fit(cumclf.X_calib, cumclf.y_calib, image_input=True,)
+    with pytest.raises(ValueError, match=r"Invalid X.*"):
+        mapie.predict(X_wrong_image)
+
+
+def test_undifined_model() -> None:
```
Comment (gmartinonQM) [inline mapie/tests/test_classification.py:367]: typo: undefined

File: mapie/tests/test_classification.py [hunk 4]
```diff
@@ -643,6 +741,34 @@ def test_cumulated_scores() -> None:
...
 
 
+@pytest.mark.parametrize("X", IMAGE_INPUT)
+def test_image_cumulated_scores(X: np.ndarray) -> None:
+    """Test image as input for cumulated_score method."""
+    alpha = [0.65]
+    quantile = [0.750183952461055]
+    # fit
+    X_calib = X['X_calib']
+    X_test = X['X_test']
```
Comment (gmartinonQM) [inline mapie/tests/test_classification.py:751]: Du coup `X` n'est pas un nd.array ici ?

**Review by gmartinonQM — CHANGES_REQUESTED** [review]

File: mapie/classification.py (hunk 1 above)
Comment (vincentblot28) [reply mapie/classification.py:230]: done

File: mapie/utils.py (hunk 2 above)
Comment (vincentblot28) [reply mapie/utils.py:353]: done

File: mapie/tests/test_classification.py (hunk 3 above)
Comment (vincentblot28) [reply mapie/tests/test_classification.py:367]: done

File: mapie/tests/test_classification.py (hunk 4 above)
Comment (vincentblot28) [reply mapie/tests/test_classification.py:751]: indeed, changed !

File: mapie/tests/test_classification.py [hunk 5]
```diff
@@ -742,7 +742,7 @@ def test_cumulated_scores() -> None:
 
 
 @pytest.mark.parametrize("X", IMAGE_INPUT)
-def test_image_cumulated_scores(X: np.ndarray) -> None:
+def test_image_cumulated_scores(X: dict) -> None:
```
Comment (gmartinonQM) [inline mapie/tests/test_classification.py:745]: ```Dict[str, np.ndarray``` (à vérifier)

**Review by gmartinonQM — CHANGES_REQUESTED** [review]

File: mapie/tests/test_classification.py (hunk 5 above)
Comment (vincentblot28) [reply mapie/tests/test_classification.py]: indeed ! code corrected

**Pull request merged**